### ISSUE TITLE
Wishlist Component cart not found

### ISSUE THREAD

**Issue by bmiskoski** [OFFLINE-GmbH/oc-mall-plugin#647]
Greetings,

I'm trying to implement the wish list functionality on a project. 

I'm getting two errors, the first is when i place the wish list button component it gives me empty product_id (it can be fixed by adding item before product_id in item.htm in wishlistbutton component)

From this: product_id: '{{ __SELF__.encode(product_id) }}
To this: product_id: '{{ __SELF__.encode(item.product_id) }}

After that i can add products in wish list but I'm getting: 
**An exception has been thrown during the rendering of a template ("The component 'cart' is not found.").**

![image](https://user-images.githubusercontent.com/58108005/100242351-fc24b100-2f34-11eb-81c5-69bc0bd0c634.png)

If i change {% partial 'cart::table/header' %} to:

{% component cart ~ '::table/header' %} and to all cases in the code the error is gone but the cart with the wishlisted products is not showing.

Tried to implement the button on a fresh installation I get the same error.

**Comment by tobias-kuendig** [comment]
Do you have the cart component on your page or layout? Did you compare your setup with the demo theme?

**Comment by bmiskoski** [comment]
I compared the setup to the demo theme, I didn't include the the cart component because it wasn't in the demo theme, however I added the cart component two days ago and I sorted the problem, also I put the js logic in {% put scripts %} tags because it rendered before my jquery CDN and didn't work for the addition of removal of products from the wishlist.

Now I have a problem because the addition of products or their removal from the wishlist is completed only once and afterwards is doesn't work, I'll rewrite some of the logic and hopefully can sort it on my own.

I have a request if maybe the backend files of the demo theme can be visible so that we as the users can search through them and won't waste your time with questions.
